### PR TITLE
MAISTRA-1717 Don't install yq-python, update yq to v4

### PR DESCRIPTION
**Pull request by dgn**
Update Dockerfile based on changes introduced in https://github.com/maistra/istio-operator/pull/682

### CODE REVIEW

**Comment by luksa** [comment]
The operator now no longer requires the yq binary, as it pulls it in the sources and builds it by itself. Also, I believe that the Istio build requires the go-based yq version 3.x. The 3.x and 4.x versions are incompatible.

**Review by jwendell — COMMENTED** [review]
IIRC the docs also use yq. @brian-avery you confirm these changes are good?

**Comment by dgn** [comment]
> The operator now no longer requires the yq binary, as it pulls it in the sources and builds it by itself. 

In that case we should probably use the same logic in the maistra.github.io repo.

> Also, I believe that the Istio build requires the go-based yq version 3.x. The 3.x and 4.x versions are incompatible.

AFAIK it doesn't. The only places we use it are istio-operator, the website and here in test-infra to do some yaml mangling

**Comment by luksa** [comment]
> > Also, I believe that the Istio build requires the go-based yq version 3.x. The 3.x and 4.x versions are incompatible.
> 
> AFAIK it doesn't. The only places we use it are istio-operator, the website and here in test-infra to do some yaml mangling

Here's where Istio upstream uses the go-based yq version 3.x:
https://github.com/istio/istio/blob/master/Makefile.core.mk#L367-L368

Maistra currently does not have this (yet).

File: docker/maistra-builder_2.1.Dockerfile [hunk 1]
```diff
@@ -47,7 +47,7 @@ RUN curl -sfL https://download.docker.com/linux/fedora/docker-ce.repo -o /etc/yu
 ENV GOBIN=/usr/local/bin
 RUN GO111MODULE=off go get github.com/myitcv/gobin && \
     gobin github.com/jstemmer/go-junit-report@${GO_JUNIT_REPORT_VERSION} && \
-    gobin github.com/mikefarah/yq/v3 && \
+    gobin github.com/mikefarah/yq/v4 && \
```
Comment (brian-avery) [inline docker/maistra-builder_2.1.Dockerfile:50]: I think @luksa said there were issues created by changes between v3 and v4 right? Have those been fixed already?

**Comment by dgn** [comment]
I can't believe they're breaking their CLI API every version.

In any case, we'll need to change maistra.github.io to merge this. Or we just leave things as they are

**Comment by brian-avery** [comment]
It looks like their last breaking changes were in v2 back in January 2020. So upgrade once a year? Doesn't sound too bad. I can play around and see what it takes to upgrade Maistra.io. I don't think it should take much

**Comment by dgn** [comment]
This was solved a while ago. Closing

**Pull request closed without merging**